docs/project/os-onboarding.md

```diff
@@ -0,0 +1,53 @@
+# OS Onboarding Guide
+
+Adding support for new operating systems (largely just new versions) is a frequent need. This guide describes how we do that, including policies we use.

```

Comment:
Does this apply for new OS families, distros, architectures too, or just new versions of existing distros of the same architecture?  Seems to me like you are trying to write a general document but it's only covering a narrow scenario.

Reply:
Most of what we do is add support for new versions. When we added new architectures, we followed the same pattern. For example, when we added support for Azure Linux, it was the same pattern. While that example is a bit cheating (since it was Mariner re-branded), we still have to do all the things we'd do if it was completely new to us.

Reply:
Adding new RIDs is no longer a thing, but pretty sure we have more to do here around new versions of apple, new distros, architecture, mobile / wasm, etc.  I'm fine with this doc not covering that - just call out what is out of scope.

Reply:
This doc is intented to cover the typical cases. I think you'll find that there is very little we have to do beyond testing. Perhaps I'm mistaken.

Reply:
I think that the doc title should make it clear that this doc really only covers new OS versions.

https://github.com/dotnet/runtime/pull/112026/files#r1938542342

If somebody from the community wanted to onboard actual new OS, this doc would not be applicable.